### PR TITLE
Add project & many licenses & ignore

### PR DESCRIPTION
**Pull request by EldersJavas**
### Add project & many licenses & ignore

Hello,

  I try to add some new licenses, and I found bsd-4 has a [project] blank, so I added it.

Good Luck!
  
EldersJavas

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -134,3 +138,14 @@ func getYear() string {
...
+
+func getProject() string {
+	if fProject != "" {
+		return fProject
+	}
+	rootDir, err := filepath.Abs(filepath.Dir(os.Args[0]))
+	if err != nil {
+		return ""
+	}
+	return strings.Split(strings.ReplaceAll(rootDir, "\\", "/"), "/")[len(strings.Split(strings.ReplaceAll(rootDir, "\\", "/"), "/"))-1]
```
Comment (nishanths) [inline main.go:150]: Would it be possible to replace this line with `filepath.Base(rootDir)`?

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,101 @@
+### Go template
```
Comment (nishanths) [inline .gitignore:1]: Please remove this file. We can discuss it an issue and the maybe add it in a separate PR.

File: .templates/zlib.tmpl [hunk 3]
```diff
@@ -0,0 +1,19 @@
+zlib License
+
+C {{.Year}} {{.Name}}
```
Comment (nishanths) [inline .templates/zlib.tmpl:3]: https://www.zlib.net/zlib_license.html

The license text at the above URL uses the full word "Copyright". Can you make this change?

File: README.md [hunk 4]
```diff
@@ -50,27 +67,30 @@ yay -S nishanths-license-git
 
 To print a license to stdout, run the `license` command followed by the license name:
 
-```
+```powershell
```
Comment (nishanths) [inline README.md:70]: Please remove the powershell syntax for the code blocks, or replace them with just `sh`.

File: main.go [hunk 5]
```diff
@@ -134,3 +138,14 @@ func getYear() string {
 	}
 	return strconv.Itoa(time.Now().Year())
 }
+
+func getProject() string {
+	if fProject != "" {
+		return fProject
+	}
+	rootDir, err := filepath.Abs(filepath.Dir(os.Args[0]))
```
Comment (nishanths) [inline main.go:146]: The updated documentation in the README says:
> The current folder is used as the project name if `-project` is omitted.

Based on this, shouldn't this code use `os.Getwd`, not `os.Args[0]`?

**Review by nishanths — COMMENTED** [review]
Thanks for the change! I added a few review comments.

**Comment by EldersJavas** [comment]
Got it, thanks, I will edit tonight

**Comment by EldersJavas** [comment]
OK, finished

File: main.go [hunk 6]
```diff
@@ -134,3 +138,18 @@ func getYear() string {
...
+		return fProject
+	}
+	dir, err := os.Getwd()
+	if err != nil {
+		return ""
+	}
+	rootDir, err := filepath.Abs(dir)
+	if err != nil {
+		return ""
+	}
```
Comment (nishanths) [inline main.go:153]: The os.Getwd documentation says it returns a rooted path (which, I think, means absolute path), so calling filepath.Abs on it should not be necessary. Anyway, no harm in having it.

**Review by nishanths — APPROVED** [review]

**Comment by nishanths** [comment]
Looks good, thank you again.

**Comment by nishanths** [comment]
I've pushed a commit to fix the lint checks.

**Comment by EldersJavas** [comment]
Ok, thanks

**Comment by nishanths** [comment]
Actually, I couldn't push to the PR branch. So I'll merge this and make a separate commit for the lint check. Thank you!

**Comment by nishanths** [comment]
Added a new version tag that includes these changes.

**Pull request merged**